inference-engine/tests/functional/plugin/shared/include/behavior/infer_request_cancellation.hpp

```diff
@@ -0,0 +1,151 @@
+#include "functional_test_utils/blob_utils.hpp"
+#include "ngraph_functions/pass/convert_prc.hpp"
+#include "ngraph_functions/subgraph_builders.hpp"
+#include "behavior/infer_request_cancellation.hpp"
+
+namespace BehaviorTestsDefinitions {
+using CancellationTests = BehaviorTestsUtils::BehaviorTestsBasic;
+
+TEST_P(CancellationTests, canCancelAsyncRequest) {
+    // Skip test according to plugin specific disabledTestPatterns() (if any)
+    SKIP_IF_CURRENT_TEST_IS_DISABLED()
+    // Create CNNNetwork from ngrpah::Function
+    InferenceEngine::CNNNetwork cnnNet(function);
+    // Load CNNNetwork to target plugins
+    auto execNet = ie->LoadNetwork(cnnNet, targetDevice, configuration);
+    // Create InferRequest
+    InferenceEngine::InferRequest req = execNet.CreateInferRequest();
+    bool cancelled = false;
+    req.SetCompletionCallback<std::function<void(InferenceEngine::InferRequest, InferenceEngine::StatusCode)>>(
+            [&](InferenceEngine::InferRequest request, InferenceEngine::StatusCode status) {
+                if (targetDevice == CommonTestUtils::DEVICE_CPU) {
+                    cancelled = (status == InferenceEngine::StatusCode::INFER_CANCELLED);
+                }
+            });
+
+    req.StartAsync();
+    InferenceEngine::StatusCode cancelStatus = req.Cancel();
+    InferenceEngine::StatusCode waitStatus = req.Wait(InferenceEngine::IInferRequest::WaitMode::RESULT_READY);
+
+    if (targetDevice == CommonTestUtils::DEVICE_CPU) {
+        ASSERT_EQ(true, cancelStatus == InferenceEngine::StatusCode::OK ||
+                        cancelStatus == InferenceEngine::StatusCode::INFER_NOT_STARTED);
+        if (cancelStatus == InferenceEngine::StatusCode::OK) {
+            ASSERT_EQ(true, cancelled);
+            ASSERT_EQ(static_cast<int>(InferenceEngine::StatusCode::INFER_CANCELLED), waitStatus);
+        } else {
+            ASSERT_EQ(false, cancelled);
+            ASSERT_EQ(static_cast<int>(InferenceEngine::StatusCode::OK), waitStatus);
+        }
+    } else {
+        ASSERT_EQ(static_cast<int>(InferenceEngine::StatusCode::NOT_IMPLEMENTED), cancelStatus);
+        ASSERT_EQ(static_cast<int>(InferenceEngine::StatusCode::OK), waitStatus);
+    }
+}
+
+TEST_P(CancellationTests, canResetAfterCancelAsyncRequest) {
+    // Skip test according to plugin specific disabledTestPatterns() (if any)
+    SKIP_IF_CURRENT_TEST_IS_DISABLED()
+    // Create CNNNetwork from ngrpah::Function
+    InferenceEngine::CNNNetwork cnnNet(function);
+    // Load CNNNetwork to target plugins
+    auto execNet = ie->LoadNetwork(cnnNet, targetDevice, configuration);
+    // Create InferRequest
+    InferenceEngine::InferRequest req = execNet.CreateInferRequest();
+
+    req.StartAsync();
+    InferenceEngine::StatusCode cancelStatus = req.Cancel();
+    InferenceEngine::StatusCode waitStatus = req.Wait(InferenceEngine::IInferRequest::WaitMode::RESULT_READY);
+
+    if (targetDevice == CommonTestUtils::DEVICE_CPU) {
```

Comment:
checking the status here is redundant (as the test as far as I can see tests for the ability to "restart" the request after cancelation). Otherwise, (if you still want to test the status  you have to impl logic like in prev test to avoid race between Infer completion and Cancel) , but this would be clear duplication